ts/utils/strings.ts

```diff
@@ -30,19 +30,30 @@ export function isTextIncludedCaseInsensitive(
  * @param text
  * @param separator
  */
-export function capitalize(text: string, separator: string = " ") {
-  return text
+export function capitalize(text: string, separator: string = " "): string {
+  // Match leading and trailing spaces
+  const leadingSpacesMatch = /^\s*/.exec(text);
+  const trailingSpacesMatch = /\s*$/.exec(text);
+
+  const leadingSpaces = leadingSpacesMatch ? leadingSpacesMatch[0] : "";
+  const trailingSpaces = trailingSpacesMatch ? trailingSpacesMatch[0] : "";
+
+  // Capitalize the words between the separators
+  const capitalizedText = text
+    .trim() // Remove leading/trailing spaces for processing
     .split(separator)
-    .reduce(
-      (acc: string, curr: string, index: number) =>
-        `${acc}${index === 0 ? "" : separator}${curr.replace(
-          new RegExp(curr.trimLeft(), "ig"),
-          _.capitalize(curr.trimLeft())
-        )}`,
-      ""
-    );
-}
+    .map(token =>
+      // Handle words with apostrophes
+      token
+        .split("'")
+        .map(subToken => _.upperFirst(subToken.toLowerCase()))
+        .join("'")
+    )
+    .join(separator);
```

Comment:
Since `capitalize` is a common utility function used by other domains to capitalize the first letter of each word (as in the previous implementation), what do you think about creating a separate function, perhaps called `capitalizeTextName` (or something like that), specifically for this use-case?

That way, we can keep the original capitalize function for basic capitalization and use `capitalizeTextName` for handling apostrophes and spaces. This will help maintain backwards compatibility and avoid any unexpected behavior in other parts of the code.

Additionally, it would be helpful to document the behavior of this new function using JSDoc in the comment. That way, anyone using it in the future will understand its specific purpose and how it differs from the basic capitalize function.

Let me know what do you think about!

Reply:
Sure I like that approach! 

Just to be on the same page:
- Revert the capitalize function to its original implementation (as it was before this PR).
- Introduce a new function named `capitalizeTextName`, which will contain the changes I introduced in this PR.

Let me know if this sounds good!

Reply:
if so, update CgnDetailScreen to use this new function in the Card title and the "nome" and "cognome" fields